README.md

```diff
@@ -45,15 +45,58 @@ On Unix-like systems, you can also copy the `dev.py` script to the root of your
 
 ## 🧪 Custom commands
 
-`devpy` can invoke custom commands. These commands define their own arguments, and they also have access to the `pyproject.toml` file for further configuration.
+`devpy` can invoke custom commands. These commands define their own arguments, and have access to the `pyproject.toml` file for further configuration.
 
-See, e.g., the [example custom command](https://github.com/scientific-python/devpy/blob/main/custom/__init__.py).
+See, e.g., the [example custom command](https://github.com/scientific-python/devpy/blob/main/example_pkg/.devpy/cmds.py).
 
 Add custom commands to the `commands` variable in the `[tool.devpy]` section of `pyproject.toml` as follows:
 
 ```
-commands = [..., 'custom/__init__.py:example']
+commands = [..., '.devpy/cmds.py:example']
 ```
 
-Here, the command is stored in `custom/__init__.py`, and the function
+Here, the command is stored in `.devpy/cmds.py`, and the function
 is named `example`.
+
+### Configuration
+
+Custom commands can access the `pyproject.toml` as follows:
+
+```python
+from devpy import util
+
+@click.command()
+def example():
+    """Command that accesses `pyproject.toml` configuration"""
+    config = util.get_config()
+    print(config["tool.devpy"])
+```
+
+### Command sections
+
+Once you have several commands, it may be useful to organize them into sections.
+In `pyproject.toml`, instead of specifying the commands as a list, use the following structure:
+
+```toml
+[tool.devpy.commands]
+"Build" = ["devpy.build", "devpy.test"]
+"Environments" = ["devpy.shell", "devpy.ipython", "devpy.python"]
+```
+
+These commands will then be rendered as:
+
+```
+Build:
+  build  🔧 Build package with Meson/ninja and install
+  test   🔧 Run tests
+
+Environments:
+  shell    💻 Launch shell with PYTHONPATH set
+  ipython  💻 Launch IPython shell with PYTHONPATH set
+  python   🐍 Launch Python shell with PYTHONPATH set
+```
+
+## History
+
+The `dev.py` tool was [proposed for SciPy](https://github.com/scipy/scipy/issues/15489) by Ralf Gommers, [implemented](https://github.com/scipy/scipy/pull/15959) by Sayantika Banik, Eduardo Naufel Schettino, and Ralf Gommers, and [improved](https://github.com/scipy/scipy/commit/8704c0025539bd97473da8b84c8f0735c31ff6e2) by Smit Lunagariya, Gagandeep Singh, Matthew Brett, Dhruv Vats, and Harsh Mishra.
```

Comment:
That's pretty close I'd say. You can probably leave out the "improved" bit - those were all very tiny tweaks, no new features or significant changes I believe.

https://labs.quansight.org/blog/the-evolution-of-the-scipy-developer-cli is a blog post that describes more about the implementation, may be worth linking.